app/ts/components/AmountField.tsx

```diff
@@ -38,18 +44,31 @@ export const AmountField = (props: Props) => {
 		}
 	}
 
+	const isCalculatingMax = useComputed(() => tokenBalance.value.state === 'pending')
+
+	const handleMax = () => {
```

Comment:
maybe some better function name than `handle` as its quite vague on what it does? I am not sure what this function does. I guess it updates the token balance to the max?

Reply:
changed to `setTokenBalanceAsAmount` to describe what it does.